problem_2/lib/even_fibonacci_numbers.rb

```diff
@@ -0,0 +1,21 @@
+#By considering the terms in the Fibonacci sequence whose 
+#values do not exceed four million, 
+#find the sum of the even-valued terms.
+
+class FibonacciCalculator
+
+	def initialize(limit)
+		@limit = limit
+	end
+
+	def add_fib_sequence
+		fib = [0, 1]
+		place = 0
+			while (fib[place] + fib[place + 1]) < @limit do
```

Comment:
Not sure if I'm misunderstanding the problem, but doesn't the problem mean if the term is below 4 million sum it (if it's not even) and if the term is above 4 million don't include it. So I think this should be `fib[place + 1] < @limit`.

Reply:
That's the problem as I understand it, but without adding `fib[place]` to `fib[place + 1]` it will sometimes append the next biggest element which will be over the `@limit`. Without that the first test (even elements of the sequence under 100) returns `[0, 2, 8, 34, 144]`.

Reply:
Ok yeah I understand it now. It'll return `144` as part of the sequence because the sum of `fib[place] + fib[place + 1]`. Alternatively it could be written like:
```
  def add_fib_sequence
    fib = [1, 2]
    place = 0
    while fib[place + 1] < @limit do
      if fib[place] + fib[place + 1] < @limit
        fib << fib[place] + fib[place + 1]
        place += 1
      end
    end
    fib.delete_if {|digit| digit % 2 != 0 }
    fib.sum
  end
```

Although your way is more succinct